node/src/run_modes.rs

```diff
@@ -428,115 +655,144 @@ parm2 - msg2\n"
-            .expect_privilege_result(false);
-        subject.privilege_dropper = Box::new(privilege_dropper);
-        let mut initialization_h_holder = FakeStreamHolder::new();
-        let mut initialization_v_holder = FakeStreamHolder::new();
-        let mut service_mode_h_holder = FakeStreamHolder::new();
-        let mut service_mode_v_holder = FakeStreamHolder::new();
-
-        let initialization_h_exit_code = subject.go(
-            &["--initialization".to_string(), "--help".to_string()],
-            &mut initialization_h_holder.streams(),
+    fn is_help_or_version_works() {
+        vec![
+            &convert_str_vec_slice_into_vec_of_strings(&["whatever", "--help", "something"]),
+            &convert_str_vec_slice_into_vec_of_strings(&["whatever", "--version", "something"]),
+            &convert_str_vec_slice_into_vec_of_strings(&["whatever", "-V", "something"]),
+            &convert_str_vec_slice_into_vec_of_strings(&["whatever", "-h", "something"]),
+        ]
+        .into_iter()
+        .for_each(|args| assert!(RunModes::is_help_or_version(args)))
+    }
+
+    #[test]
+    fn daemon_and_node_modes_help_call() {
+        let subject = RunModes::new();
+        let mut daemon_h_holder = FakeStreamHolder::new();
+        let mut node_h_holder = FakeStreamHolder::new();
+
+        let daemon_h_exit_code = subject.go(
+            &[
+                "program".to_string(),
+                "--initialization".to_string(),
+                "--help".to_string(),
+            ],
+            &mut daemon_h_holder.streams(),
         );
-        let initialization_v_exit_code = subject.go(
-            &["--initialization".to_string(), "--version".to_string()],
-            &mut initialization_v_holder.streams(),
+
+        let node_h_exit_code = subject.go(
+            &["program".to_string(), "--help".to_string()],
+            &mut node_h_holder.streams(),
         );
-        let service_mode_h_exit_code = subject.go(
-            &["--help".to_string()],
-            &mut service_mode_h_holder.streams(),
+
+        assert_eq!(daemon_h_exit_code, 0);
+        let daemon_stdout_message = daemon_h_holder.stdout.get_string();
+        assert!(daemon_stdout_message.contains("MASQ\nMASQ Node is the foundation of  MASQ Network, an open-source network that allows anyone to"));
+        assert!(daemon_stdout_message.contains("--initialization    Directs"));
+        assert_eq!(daemon_h_holder.stderr.get_string(), "");
+        assert_eq!(node_h_exit_code, 0);
+        let node_stdout_message = node_h_holder.stdout.get_string();
+        assert!(node_stdout_message
+            .contains("to allocate spare computing\nresources to make the internet a free"));
+        assert!(node_stdout_message.contains("--clandestine-port <CLANDESTINE-PORT>\n"));
+        assert_eq!(node_h_holder.stderr.get_string(), "");
+    }
+
+    #[ignore] //remove this attribute when Clap is more stable or transform this test into an integration test
```

Comment:
This test can't work, because `clap` doesn't work correctly for --version with `.get_matches_from_safe()`.